### PR TITLE
feat: update default scheduler to priority for agentrm

### PR DESCRIPTION
**Pull request by kkunapuli**
## Ticket

[RM-254]

## Description

For agent resource managers, use the priority scheduler by default.

## Test Plan

Start a new instance of Determined, e.g. by running devcluster, with no configuration. On the cluster overview page, the resource pool description should say Scheduler Type Priority.

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[RM-254]: https://hpe-aiatscale.atlassian.net/browse/RM-254?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/cmd/determined-master/root.go [hunk 1]
```diff
@@ -193,18 +193,21 @@ func applyBackwardsCompatibility(configMap map[string]interface{}) (map[string]i
...
+	oldRMConfig := schedulerExisted || provisionerExisted
+	newRMConfig := rmExisted || rpsExisted
+	if newRMConfig && oldRMConfig {
 		return nil, errors.New(
 			"cannot use the old and the new configuration schema at the same time",
 		)
 	}
-	if rmExisted || rpsExisted {
+	if !oldRMConfig {
+		// Use configMap if RMs are not defined at all, or if they are defined using the new schema.
```
Comment (kkunapuli) [inline master/cmd/determined-master/root.go:204]: I have mixed feelings on this change. I think it's the right thing to do, but I'm suspicious of unintended consequences.

Comment (stoksc) [reply master/cmd/determined-master/root.go:204]: i agree on both fronts. i would just run a cluster before and after this change, call `det master config` for both, and diff the results. then maybe post it here if there is anything concerning.

Comment (kkunapuli) [reply master/cmd/determined-master/root.go:204]: Thanks for the suggestion! There's only one line different between the configs with/without this specific change. The "new" / proposed version has the addition of this line:
```max_cpu_containers_per_agent: -1```

Which looks like it's probably fine to me.

Comment (stoksc) [reply master/cmd/determined-master/root.go:204]: Yeah, the -1 sentinel value seems right to me.

**Review by stoksc — APPROVED** [review]

File: docs/release-notes/feature-clean-up.rst [hunk 2]
```diff
@@ -10,10 +10,10 @@ Agent Resource Manager:
...
 -  Container Runtimes: We will limit support to Docker for Agent Resource Managers.
 
--  Job Scheduling: Moving a job will require adjusting its priority; jobs cannot be shifted within
-   the same priority group. Support for round-robin and fair share schedulers is discontinued. We
-   recommend using the priority scheduler, as it meets most scheduling needs and simplifies
-   configuration.
+-  Job Scheduling: Switching to ``priority`` for the default scheduler. Support for round-robin and
+   fair share schedulers is discontinued. We recommend using the priority scheduler, as it meets
+   most scheduling needs and simplifies configuration. Moving a job will require adjusting its
+   priority; jobs cannot be shifted within the same priority group.
```
Comment (tara-det-ai) [inline docs/release-notes/feature-clean-up.rst:16]: -  Job Scheduling: The default scheduler is now ``priority``. Support for the round-robin and fair share schedulers has been discontinued. We recommend using the priority scheduler, as it meets
   most scheduling needs and simplifies configuration. To move a job, you will need to adjust its priority; jobs cannot be shifted within the same priority group.

**Pull request merged**